src/order/rel_iso.lean

```diff
@@ -147,33 +155,68 @@ end
 @[simp] theorem of_monotone_coe [is_trichotomous α r] [is_asymm β s] (f : α → β) (H) :
   (@of_monotone _ _ r s _ _ f H : α → β) = f := rfl
 
--- If le is preserved by an relation embedding of preorders, then lt is too
-def lt_embedding_of_le_embedding [preorder α] [preorder β]
-  (f : ((≤) : α → α → Prop) ↪r ((≤) : β → β → Prop)) :
-(has_lt.lt : α → α → Prop) ↪r (has_lt.lt : β → β → Prop) :=
-{ map_rel_iff' := by intros; simp [lt_iff_le_not_le,f.map_rel_iff], .. f }
+/-- Embeddings of partial orders that preserve  -/
+def order_embedding_of_lt_embedding [partial_order α] [partial_order β]
+  (f : ((<) : α → α → Prop) ↪r ((<) : β → β → Prop)) :
+  α ↪o β :=
+{ map_rel_iff' := by { intros, simp [le_iff_lt_or_eq,f.map_rel_iff, f.injective] }, .. f }
 
 end rel_embedding
 
-/-- The inclusion map `fin n → ℕ` is an relation embedding. -/
-def fin.val.rel_embedding (n) : @rel_embedding (fin n) ℕ (<) (<) :=
+namespace order_embedding
+
+variables [preorder α] [preorder β] (f : α ↪o β)
+
+/-- lt is preserved by order embeddings of preorders -/
+def lt_embedding : ((<) : α → α → Prop) ↪r ((<) : β → β → Prop) :=
+{ map_rel_iff' := by intros; simp [lt_iff_le_not_le,f.map_rel_iff], .. f }
```

Comment:
Could you add a `simp` lemma of the form `lt_embedding_apply : lt_embedding f x = f x`?